listzones.go

```diff
@@ -0,0 +1,80 @@
+package cfdns
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/simplesurance/cfdns/logs"
+)
+
+// Listzones lists zones on CloudFlare.
+//
+// API Reference: https://developers.cloudflare.com/api/operations/zones-get
+func (c *Client) ListZones(
+	_ *ListZonesRequest,
+) *Iterator[ListZonesResponseItem] {
+	page := 0
+
+	return &Iterator[ListZonesResponseItem]{
+		fetchNext: func(
+			ctx context.Context,
+		) ([]*ListZonesResponseItem, bool, error) {
+			page++
+
+			queryParams := url.Values{
+				"direction": {"asc"},
+				"per_page":  {strconv.Itoa(itemsPerPage)},
+			}
+
+			if page != 0 {
+				queryParams["page"] = []string{strconv.Itoa(page)}
+			}
+
+			resp, err := runWithRetry[*listZoneAPIResponse](
+				ctx,
+				c,
+				c.logger.SubLogger(logs.WithPrefix("ListZones"), logs.WithInt("page", page)),
+				&request{
+					method:      http.MethodGet,
+					path:        "zones",
+					queryParams: queryParams,
+					body:        nil,
+				})
+
+			if err != nil {
+				return nil, false, err
+			}
+
+			items := make([]*ListZonesResponseItem, len(resp.body.Result))
+			for i, v := range resp.body.Result {
+				items[i] = &ListZonesResponseItem{
+					ID:   v.ID,
+					Name: v.Name,
+				}
+			}
+
+			isLast := len(resp.body.Result) < itemsPerPage
```

Comment:
The endpoint might return less items then requested because e.g. it has a lower itemsPerPage limit. If that happen isLast would be true despite not all items were fetched.

Using the returned `total_count` value to determine if we are at the last page seems much safer to me.